src/avif.c

```diff
@@ -182,6 +182,40 @@ void avifImageCopyNoAlloc(avifImage * dstImage, const avifImage * srcImage)
     dstImage->imir = srcImage->imir;
 }
 
+void avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
+{
+    assert((srcImage->width == dstImage->width) && (srcImage->height == dstImage->height) && (srcImage->depth == dstImage->depth) &&
+           (srcImage->yuvFormat == dstImage->yuvFormat) && (srcImage->yuvRange == dstImage->yuvRange));
```

Comment:
This assertion fails in Debug build. The following patch fixes the assertion failure:

```
diff --git a/src/avif.c b/src/avif.c
index 79e5082..fd6ef9c 100644
--- a/src/avif.c
+++ b/src/avif.c
@@ -184,8 +184,7 @@ void avifImageCopyNoAlloc(avifImage * dstImage, const avifImage * srcImage)
 
 void avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
 {
-    assert((srcImage->width == dstImage->width) && (srcImage->height == dstImage->height) && (srcImage->depth == dstImage->depth) &&
-           (srcImage->yuvFormat == dstImage->yuvFormat) && (srcImage->yuvRange == dstImage->yuvRange));
+    assert(srcImage->depth == dstImage->depth);
     const size_t bytesPerPixel = avifImageUsesU16(srcImage) ? 2 : 1;
 
     const avifBool skipColor = !(planes & AVIF_PLANES_YUV);
@@ -206,6 +205,8 @@ void avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avif
         if (!srcRow) {
             continue;
         }
+        assert(planeWidth == avifImagePlaneWidth(dstImage, c));
+        assert(planeHeight == avifImagePlaneHeight(dstImage, c));
 
         const size_t planeWidthBytes = planeWidth * bytesPerPixel;
         for (uint32_t y = 0; y < planeHeight; ++y) {
```

This patch moves the checks for width and height back inside the for loops. To be conservative, it asserts srcImage and dstImage have the same width and height for the plane only when it is about to copy that plane.

Also, this patch doesn't check if yuvRange is the same when we need to copy YUV planes. That check is less important because if it is not true, we still won't have memory errors.

An alternative fix is to write the assertions according to the comment that I suggested:

```
If the AVIF_PLANES_YUV bit is set in `planes`, then srcImage and dstImage must have the same yuvFormat and yuvRange.
```